posthog-react-native/src/surveys/components/QuestionTypes.tsx

```diff
@@ -0,0 +1,353 @@
+import React, { useMemo, useState } from 'react'
+import { Pressable, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native'
+
+import {
+  CheckSVG,
+  DissatisfiedEmoji,
+  NeutralEmoji,
+  SatisfiedEmoji,
+  VeryDissatisfiedEmoji,
+  VerySatisfiedEmoji,
+} from '../icons'
+import { getContrastingTextColor, getDisplayOrderChoices, SurveyAppearanceTheme } from '../surveys-utils'
+import {
+  BasicSurveyQuestion,
+  LinkSurveyQuestion,
+  MultipleSurveyQuestion,
+  RatingSurveyQuestion,
+  SurveyRatingDisplay,
+  SurveyQuestionType,
+} from '../../../../posthog-core/src/surveys-types'
+import { BottomSection } from './BottomSection'
+import { QuestionHeader } from './QuestionHeader'
+
+export function OpenTextQuestion({
+  question,
+  appearance,
+  onSubmit,
+}: {
+  question: BasicSurveyQuestion
+  appearance: SurveyAppearanceTheme
+  onSubmit: (text: string) => void
+}): JSX.Element {
+  const [text, setText] = useState('')
+
+  return (
+    <View>
+      <QuestionHeader
+        question={question.question}
+        description={question.description}
+        descriptionContentType={question.descriptionContentType}
+      />
+      <View style={styles.textInputContainer}>
+        <TextInput
+          style={styles.textInput}
+          multiline
+          numberOfLines={4}
+          placeholder={appearance.placeholder}
+          onChangeText={setText}
+          value={text}
+        />
+      </View>
+      <BottomSection
+        text={question.buttonText ?? appearance.submitButtonText}
+        submitDisabled={!text && !question.optional}
+        appearance={appearance}
+        onSubmit={() => onSubmit(text)}
+      />
+    </View>
+  )
+}
+
+export function LinkQuestion({
+  question,
+  appearance,
+  onSubmit,
+}: {
+  question: LinkSurveyQuestion
+  forceDisableHtml: boolean
+  appearance: SurveyAppearanceTheme
+  onSubmit: (clicked: string) => void
+}): JSX.Element {
+  return (
+    <>
+      <QuestionHeader
+        question={question.question}
+        description={question.description}
+        descriptionContentType={question.descriptionContentType}
+      />
+      <BottomSection
+        text={question.buttonText ?? appearance.submitButtonText ?? 'Submit'}
+        submitDisabled={false}
+        link={question.link}
+        appearance={appearance}
+        onSubmit={() => onSubmit('link clicked')}
+      />
+    </>
+  )
+}
+
+export function RatingQuestion({
+  question,
+  appearance,
+  onSubmit,
+}: {
+  question: RatingSurveyQuestion
+  appearance: SurveyAppearanceTheme
+  onSubmit: (rating: number | null) => void
+}): JSX.Element {
```

Comment:
you can probably combine the props into a common interface so you don't have to repeat the same stuff. might make it easier to not have to cast cast as `any` when calling those components

[example from js-sdk](https://github.com/PostHog/posthog-js/blob/main/src/extensions/surveys/components/QuestionTypes.tsx#L25-L40)

not really needed but nice to have if we add more props in the future

Reply:
good point [`d75391d` (#422)](https://github.com/PostHog/posthog-js-lite/pull/422/commits/d75391dc6e67d3a3f293833e5aa6f55d656ead21)